packages/api/tests/models/test-pdrs-model.js

```diff
@@ -0,0 +1,182 @@
+'use strict';
+
+const test = require('ava');
+const { randomId, randomNumber, randomString } = require('@cumulus/common/test-utils');
+
+const { deconstructCollectionId } = require('../../lib/utils');
+const Pdr = require('../../models/pdrs');
+
+let pdrsModel;
+
+test.before(async () => {
+  process.env.PdrsTable = randomString();
+  pdrsModel = new Pdr();
+  await pdrsModel.createTable();
+});
+
+test.after.always(async () => {
+  await pdrsModel.deleteTable();
+});
+
+const createPdrMessage = ({
+  collectionId = `${randomId('MOD')}___${randomNumber()}`,
+  numCompletedExecutions = 0,
+  numFailedExecutions = 0,
+  numRunningExecutions = 0,
+  timestamp = Date.now()
+} = {}) => ({
+  cumulus_meta: {
+    state_machine: randomId('pdr'),
+    execution_name: randomId('execution'),
+    workflow_start_time: timestamp
+  },
+  meta: {
+    collection: deconstructCollectionId(collectionId),
+    provider: {
+      id: 'prov1',
+      protocol: 's3',
+      host: 'random-bucket'
+    },
+    status: 'running'
+  },
+  payload: {
+    completed: [
+      ...new Array(numCompletedExecutions)
+    ].map(() => randomId('execution')),
+    failed: [
+      ...new Array(numFailedExecutions)
+    ].map(() => randomId('execution')),
+    running: [
+      ...new Array(numRunningExecutions)
+    ].map(() => randomId('execution'))
+  }
+});
+
+test('createPdrFromSns() returns undefined when no PDR name exists', async (t) => {
+  const record = await pdrsModel.createPdrFromSns(createPdrMessage());
+  t.is(record, undefined);
+});
+
+test('createPdrFromSns() throws error when meta.collection is missing', async (t) => {
+  const message = createPdrMessage();
+
+  delete message.meta.collection;
+  message.payload.pdr = {
+    name: randomId('pdr')
+  };
+
+  await t.throws(
+    () => pdrsModel.createPdrFromSns(message),
+    { instanceOf: TypeError }
```

Comment:
Perhaps it should throw something more meaningful than a `TypeError`?

Reply:
I actually removed this test after the refactor to make all these model methods creating records from SNS handle the collection ID consistently. And I added a unit test for the new helper.